src/index.js

```diff
@@ -141,6 +141,7 @@ Cache.prototype.delete = function (key, callback) {
     if (e) return callback(new Error('Resource not found'))
     this.get(METADATA, key, (e, metadata) => {
       if (e) return callback(e)
+      if (!metadata) return callback({message: `no metadata found for ${key}`})
       metadata.status = 'deleted'
```

Comment:
Maybe better to just do:
`metadata = metadata || {}`

That way code flow proceeds as expected and a record of the deletion will exist in metadata.

Reply:
good call. keep it simple!

Reply:
Cool, just add to the changelog.